Sources/BraintreePayPal/BTPayPalClient.swift

```diff
@@ -422,32 +449,52 @@ import BraintreeDataCollector
             shopperSessionID: payPalRequest?.shopperSessionID
         )
 
-        var urlComponents = URLComponents(url: payPalAppRedirectURL, resolvingAgainstBaseURL: true)
+        var urlComponents = URLComponents(url: config.appSwitchUrl, resolvingAgainstBaseURL: true)
         urlComponents?.queryItems = [
-            URLQueryItem(name: "ba_token", value: baToken),
+            URLQueryItem(name: "ba_token", value: config.baToken),
             URLQueryItem(name: "source", value: "braintree_sdk"),
             URLQueryItem(name: "switch_initiated_time", value: String(Int(round(Date().timeIntervalSince1970 * 1000))))
         ]
         
-        guard let redirectURL = urlComponents?.url else {
+        guard let appSwitchUrl = urlComponents?.url else {
             self.notifyFailure(with: BTPayPalError.invalidURL("Unable to construct PayPal app redirect URL."), completion: completion)
             return
         }
 
-        application.open(redirectURL) { success in
-            self.invokedOpenURLSuccessfully(success, url: redirectURL, completion: completion)
+        application.open(appSwitchUrl, withOptions: [.universalLinksOnly: true]) { didOpened in
+            if didOpened {
+                self.invokedOpenURLSuccessfully(appSwitchUrl, completion: completion)
+            } else {
+                if let fallbackUrl = config.fallbackUrl {
+                    self.handleFallbackToExternalBrowser(forUrl: fallbackUrl, completion: completion)
```

Comment:
If we decide to go forward with a similar PR we should instead use `ASWebAuthenticationSession` for the fallback scenario (configured here for the non app switch flow). If our hypothesis is that `application.open` is failing to open at the OS level I do not think we should just try to open a different URL with the same method as that's likely to also fail.